<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="AY" />


<title>snakemake</title>

<script src="site_libs/header-attrs-2.19/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Seqeucing Core Facility MPI Biochemistry</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="conda.html">conda</a>
</li>
<li>
  <a href="snakemake.html">snakemake</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="contact.html">
    <span class="fa fa-envelope fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/yeroslaviz">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/AssaYeroslaviz">
    <span class="fa fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://www.linkedin.com/in/ayeroslaviz/">
    <span class="fa fa-linkedin fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">snakemake</h1>
<h4 class="author">AY</h4>
<h4 class="date">25 January, 2023</h4>

</div>


<p>This part is still work in progress.</p>
<p><code>snakemake</code> relies heavily on a <code>config</code> file,
i.e. <code>config.yaml</code>, where parameter specific for each
analysis are saved. This reduces the number of changes needed to be done
each time a new project is run.</p>
<p>For this project we are using the following <code>config</code> file:
<a href="config.yaml" target="_blank">config.yaml</a></p>
<p>The <code>config.yaml</code> file contains several parts.</p>
<ol style="list-style-type: decimal">
<li>The top one contain the information about the organism used in the
analysis. This is necessary as it provides the <em>prefix</em> for both
the genome indexing as well as the mapping steps.<br />

<ul>
<li>The <code>config</code> file contain seven different organisms and
should be run only once at the beginning to create the indexed genomes.
It must be ran again <strong>only</strong> when new organisms are added
or new versions of any genomes (or of tools) are needed. For each genome
the prefix would be the three-letter name of the organism followed by
the genome built version (e.g <em>Hsp.GRCh38</em>,
<em>Cel.WBcel2358</em>).<br />
</li>
<li>Below the Organism’s name are the links for the genomic sequence in
a <code>fastA</code> format and the annotation files in a
<code>gtf</code> (gene transfer file) format. On each first time both
files will be downloaded before the genome indexing.</li>
</ul></li>
<li>The next part contains parameters for the <code>STAR</code> aligner.
In this section the parameters can be set globally and be easily called
upon, when running multiple versions of the same aligner.
<ul>
<li>The most important parameter is <code>gzipped</code>. This set
whether or not the sample files are compressed or not. <code>STAR</code>
uses an internal decompressing step on the fly, when running the mapping
step. As the fastq files from the core facility are usually
<code>gzipped</code>, the default here is set to <code>true</code>.</li>
<li>The <code>SA</code> index set the <code>preindex</code> string.
Bigger genomes take higher values. It should in general be set to <span
class="math inline">\(~log2(GenomeLength)/2-1\)</span>. For standard
genomes ( <em>hsa</em> or <em>mmu</em>) it should be set to 14, for
smaller genomes such as <em>Drosophila</em>, it should be set to 13.
<em>C. elegans</em> genome should be even smaller and be setted at 12.
The higher the value, the faster the index building, but higher is also
the memory requirements.</li>
<li>The memory can be set with the <code>RAM</code> parameter.</li>
</ul></li>
<li>The third part of the <code>config</code> file is project-related
and must be changed with each run. It has three parts:
<ul>
<li>Project number (e.g. <strong>P000</strong>)</li>
<li>Path to concatenated fastq files
(e.g. <strong>/fs/pool/pool-bcfngs/fastq_files/P000/P000_Testrun/conc.fastq/</strong>).</li>
<li>the organism to be mapped against
(e.g. <strong>Cel.WBcel235</strong>).</li>
</ul>
This parameters must be changed each time a new run is done. The run
will create a new folder within this working directory, where it will
save the mapping results in form of sorted <code>bam</code> files and
the <code>bigwig</code> files for visualiztion purposes.</li>
</ol>
<p>This <code>config</code> file as well as the snakemake scripts are
all placed in the <code>scripts/</code> folder and should stay there.
The run command will be set to find them there.</p>
<p>The <code>snakemake</code> workflow is based on so-called
<em>rules</em>. This rules are a way to split the workflow into single
steps. The advantage here, is that each time the steps can be ran
separately. If some parameters are changed to only one step in the
analysis <code>snakemake</code> can figure out which steps must be
re-run and which one don’t.</p>
<p>This snakemake protocol works in two steps.</p>
<p><strong>1. Getting the genome and indexing</strong></p>
<p><strong>2. Mapping the samples with the needed mapper and
genome</strong></p>
<p>The first more general step is to set the working directory. It would
be best, if we use as less copying/moving of files as possible. For that
reason it might be best, if we can directly work inside the
<code>conc.fastq</code> folder, where the data is created after the
demultiplxing. If so, some of the paths in the work flow would have to
be changed or maybe adding a global <code>work.dir</code> parameter to
the <code>config.yaml</code> file. (<em><code>work.dir</code> parameter
in config.yaml file must be set</em>).</p>
<div id="getting-the-genome-and-indexing" class="section level3">
<h3>Getting the genome and indexing</h3>
<p><code>This work is pending the changes when the indexed genomes will be saved on the server permanently</code></p>
<ol style="list-style-type: decimal">
<li>Should we have a folder with several genomes already indexed and
prepared for the mapping beforehand?</li>
<li>if so which genomes?</li>
<li>where should they be saved? The paths to the indexed genomes would
than needed to be changed accordingly in the second snakemake file for
the mapping step.</li>
</ol>
<p>The advantage is, that if we have this done, this first script can be
spared.</p>
<p><a href="getGenome_IndexGenome.Snakefile" target="_blank">Getting and
indexing the genome</a></p>
<p>In General the script uses the link and download both the genome and
the annotation files for the given genome. it than automatically and in
parallel indexes the genome for the three mappers - <code>star</code>,
<code>bwa</code> and <code>bowtie2</code>.</p>
<p><strong>CHANGES in version 3</strong></p>
<p>In version 3 of the script, the rule <code>get_genome</code> was
changed to accommodate the automatic download of multiple genomes. For
that reason, wildcards were added as <code>params</code>, one for the
<code>fastA</code> file and one for the <code>gtf</code> file. This is
because snakemake’s brackets markup can only replace a variable with its
string representation, but do not evaluate any codes. so first a string
should be set to be passed over to the <code>wget</code> command in the
shell.</p>
</div>
<div id="mapping-the-samples-with-the-star" class="section level3">
<h3>Mapping the samples with the <code>STAR</code></h3>
<p><a href="Star.MappingQuant.PairedEndFastq.Snakefile"
target="_blank">mapping the samples in a paired-end format</a></p>
<p>This script contains three (four) steps. in the first step the
<code>fastq</code> samples are mapped to the indexed genome. Then, they
are being indexed. Last, after creating a chromosome-size file, the
<code>bam</code> files are converted into <code>bigwig</code> files for
better visualization using the <a
href="http://genome.ucsc.edu/cgi-bin/hgTracks">UCSC browser</a>.</p>
<p>We have two separate scripts for single-end and paired-end runs. The
PE run looks for both R1 and R2 fastq files, while the SE script will
ignore all R2 samples.</p>
<p>After the environment was created <strong>and activated</strong>, one
must change to the directory of the concatenated fast files. E.g.</p>
<pre class="bash"><code>cd /fs/pool/pool-bcfngs/fastq_files/P000/P000_Testrun/conc.fastq/</code></pre>
<p>To run the script use this command to test/run. The settings are
explained below.</p>
<pre class="bash"><code># for single-end samples
snakemake -nps /fs/pool/pool-bcfngs/scripts/Star.MappingQuant.SingleEndFastq.Snakefile -j 50
# for paired-end samples
snakemake -ps /fs/pool/pool-bcfngs/scripts/Star.MappingQuant.PairedEndFastq.Snakefile -j 50</code></pre>
<ul>
<li><code>-n</code> - dry-run. This parameter will only output the
command, but execute them. This should be used to test the parameters,
but must be removed for the analysis to be done.</li>
<li><code>-p</code> - Print the commands to be execute.</li>
<li><code>-s</code> - the name of the snakeake file to be executed.</li>
<li><code>-j</code> - #cores. How many cores should be used in the
run.</li>
</ul>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
