<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="AY" />


<title>snakemake</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Seqeucing Core Facility MPI Biochemistry</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="conda.html">conda</a>
</li>
<li>
  <a href="snakemake.html">snakemake</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="contact.html">
    <span class="fa fa-envelope fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://github.com/frymor">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/AssaYeroslaviz">
    <span class="fa fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://www.linkedin.com/in/ayeroslaviz/">
    <span class="fa fa-linkedin fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">snakemake</h1>
<h4 class="author">AY</h4>
<h4 class="date">13 November, 2019</h4>

</div>


<p>This part is still work in progress.</p>
<p><code>snakemake</code> relies havealy on a config file, i.e. <code>config.yaml</code>, where parameter specific for each analysis are saved. This reduces the number of chagnes needed to be done each time a new project is run.</p>
<p>For this project we are using the following cofig file: <a href="config.yaml" target="_blank">config.yaml</a></p>
<p>The <code>config.yaml</code> file contains several parts.</p>
<ol style="list-style-type: decimal">
<li>The top one contain the nformation about hte organism used in the analysis. This is neccessary as it provides the <em>prefix</em> for both the genome indexing as well as the mapping steps. As an example it contains the genome for <em>Drosophila Melanogaster</em>. (In this workflow it is expected that each time, the genomic mapping would be done against just one genome. Therefore there is no needed for an elaboragte strucutre of multiple genomes or reading a list of links from a file)</li>
</ol>
<p>Below the Organism’s name are the links for the genomic sequence in a <code>fastA</code> format and the annotation files in a <code>gtf</code> (gene transfer file) format. For each one there is a link for the two files separately. For each run, one of the links for each part should be activ.</p>
<ol start="2" style="list-style-type: decimal">
<li>The next part contains parameters for the <code>STAR</code> aligner. In this section the parameters can be set globally and be easily called upon, when running multiple versions of the same aligner.
<ul>
<li>The most important parameter is <code>gzipped</code>. This set whether or not the sample files are compressed or not. <code>STAR</code> uses an internal decompressing step on the fly, when running the mapping step.</li>
<li>The <code>SA</code> index set the preindex string. bigger genomes can take higher values. It should in general be set to <span class="math inline">\(~log2(GenomeLength)/2-1\)</span>. For standard genomes ( <em>hsa</em> or <em>mmu</em>) it should be set to 14, for smaller genomes such as <em>Drosophila</em>, it shpuld be set to 13. <em>C. elegans</em> genome should be even smaller and be setted at 12. The higher the value, tha faster the indexbuilding, but higher is also the memory requirements.</li>
<li>The memory can be set with the <code>RAM</code> parameter.</li>
</ul></li>
<li>The last part conatin one paramer for the indexing step of the bwa algorithm. This is similar to the above <code>SA</code> parameter, but uses a so-called block size for the indexing process. Higher values here would also increases the speed of the indexing, but require more RAM.</li>
</ol>
<p>This config file must be in the same folder as the other files for the same analysis.</p>
<p>The <code>snakemake</code> workflow is based on so-called <em>rules</em>. This rules are a way to split the workflow into single steps. The advantage here, is that each time the steps can be ran separately. If some parameters are changed to only one step in the analysis <code>snakemake</code> can figure out which steps must be re-run and which one don’t.</p>
<p>This snakemake protocoll worjs in two steps.</p>
<p><strong>1. Getting the genome and indexing</strong></p>
<p><strong>2. Mapping the samples with the needed mapper and genome</strong></p>
<p>The first more general step is to set the working directory. It would be best, if we use as less copying/moving of files as possible. For that reason it might be best, if we can directly work on the <code>conc.fastq</code> folder, where the data is created after the demultiplxing. If so, some of the paths in the work flow would have to be changed or maybe adding a gobal <code>work.dir</code> parameter to the c<code>config.yaml</code> file. (<em><code>work.dir</code> parameter in config.yaml file must be set</em>).</p>
<div id="getting-the-genome-and-indexing" class="section level3">
<h3>Getting the genome and indexing</h3>
<p><code>This work is pending the changes when the indexed genomes will be saved on the server permanently</code></p>
<ol style="list-style-type: decimal">
<li>Shouled we have a folder with several genomes already indexed and prepared for the mapping beforehand?</li>
<li>if so which genomes?</li>
<li>where should they be saved? The paths to the indexed genomes would than needed to be changed accordingly in the second snakemake file for the mapping step.</li>
</ol>
<p>The advantage is, that if we have this done, this first script can be spared.</p>
<p><a href="getGenome_IndexGenome.Snakefile" target="_blank">Getting and indexing the genome</a></p>
<p>In General the script uses the link and download both the genome and the annotation files forthe given genome. it than automatically and in parallel indexes the genome for the three mappers - <code>star</code>, <code>bwa</code> and <code>bowtie2</code>.</p>
</div>
<div id="mapping-the-samples-with-the-star" class="section level3">
<h3>Mapping the samples with the <code>STAR</code></h3>
<p><a href="Star.MappingQuant.Snakefile" target="_blank">mapping the samples</a></p>
<p>This script contains three (four) steps. in the first step the <code>fastq</code> samples are mapped to the indexed genome. Then they are being indexd. and last, after creating an chromosome-size file, the bam files are converted into <code>bigwig</code> files for better visualization using the <a href="http://genome.ucsc.edu/cgi-bin/hgTracks">UCSC browser</a></p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
